controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java

```diff
@@ -1004,65 +1004,47 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
-            if (validChildType != null && !validChildType.equals(childType)) {
-                continue;
-            }
-            final ImmutableManagementResourceRegistration childRegistration = registry.getSubModel(PathAddress.pathAddress(element));
-            if (childRegistration == null) {
-                continue;
-            }
-            final AliasEntry aliasEntry = childRegistration.getAliasEntry();
+        Map<String, Set<String>> result = new HashMap<>();
 
-            Set<String> set = result.get(childType);
-            if (set == null) {
-                set = new LinkedHashSet<String>();
-                result.put(childType, set);
+        for (String childType : registry.getChildNames(PathAddress.EMPTY_ADDRESS)) {
+            if ((validChildType == null) || validChildType.equals(childType)) {
+                result.put(childType, new LinkedHashSet<>(resource.getChildrenNames(childType)));
             }
+        }
 
-            if (aliasEntry == null) {
-                if (resource != null && resource.hasChildren(childType)) {
-                    Set<String> childNames = resource.getChildrenNames(childType);
-                    if (element.isWildcard()) {
-                        set.addAll(childNames);
-                    } else if (childNames.contains(element.getValue())) {
-                        set.add(element.getValue());
-                    }
+        Set<PathElement> paths = registry.getChildAddresses(PathAddress.EMPTY_ADDRESS);
+        for (PathElement path : paths) {
+            String childType = path.getKey();
+            if ((validChildType == null) || validChildType.equals(childType)) {
+                Set<String> children = result.get(childType);
+                if (children == null) {
+                    // WFLY-3306 Ensure we have an entry for any valid child type
+                    children = new LinkedHashSet<>();
+                    result.put(childType, children);
                 }
-            } else {
-                PathAddress childAddr = addr.append(element);
-                PathAddress target = aliasEntry.convertToTargetAddress(childAddr, AliasContext.create(childAddr, context));
-                assert !childAddr.equals(target) : "Alias was not translated";
-                PathAddress targetParent = target.subAddress(0, target.size() - 1);
-                Resource parentResource = context.readResourceFromRoot(targetParent, false);
-                if (parentResource != null) {
-                    PathElement targetElement = target.getLastElement();
-                    if (targetElement.isWildcard()) {
-                        set.addAll(parentResource.getChildrenNames(targetElement.getKey()));
-                    } else if (parentResource.hasChild(targetElement)) {
-                        set.add(element.getValue());
+                ImmutableManagementResourceRegistration childRegistration = registry.getSubModel(PathAddress.pathAddress(path));
```

Comment:
I think the null check that was at L1015 before may be necessary. Unlike the Resource tree, we don't do a copy-on-write/publish-on-commit deal when an op changes the MRR tree (yet). So a read op that calls this can have an MRR disappear underneath it.